frontend/src/pages/projects/screens/detail/storage/ClusterStorageModal.tsx

```diff
@@ -130,28 +200,82 @@ const ClusterStorageModal: React.FC<ClusterStorageModalProps> = ({ existingPvc,
-            />
-          )}
-          <StorageNotebookConnections
-            setForNotebookData={(forNotebookData) => {
-              setNotebookData(forNotebookData);
-            }}
-            forNotebookData={notebookData}
-            connectedNotebooks={connectedNotebooks}
-          />
-          {restartNotebooks.length !== 0 && (
-            <FormGroup>
-              <NotebookRestartAlert notebooks={restartNotebooks} />
+          {workbenchEnabled && (
+            <FormGroup label="Workbench connections" fieldId="workbench-connections">
+              <Stack hasGutter>
+                <StackItem>
+                  <Table
+                    data-testid="workbench-connection-table"
+                    variant="compact"
+                    columns={storageColumns}
+                    data={notebookData}
+                    rowRenderer={(row, rowIndex) => (
+                      <ClusterStorageTableRow
+                        clusterStorageName={name}
+                        existingPvc={existingPvc}
+                        connectedNotebooks={connectedNotebooks}
+                        key={rowIndex}
+                        obj={row}
+                        availableNotebooks={{
+                          notebooks: [
+                            ...availableNotebooks,
+                            ...allNotebooks.filter(
+                              (currentData) => currentData.metadata.name === row.name,
+                            ),
+                          ],
+                          loaded,
+                        }}
+                        addNotebookData={(
+                          notebookName?: string,
+                          mountPath?: MountPath,
+                          isUpdatedValue?: boolean,
+                          updatedExistingPvc?: boolean,
+                        ) => {
+                          const copiedArray = [...notebookData];
+                          if (notebookName) {
+                            copiedArray[rowIndex].name = notebookName;
+                          }
+                          if (isUpdatedValue !== undefined) {
+                            copiedArray[rowIndex].isUpdatedValue = isUpdatedValue;
+                          }
+
+                          if (updatedExistingPvc) {
+                            copiedArray[rowIndex].existingPvc = updatedExistingPvc;
+                          }
+                          if (mountPath) {
+                            copiedArray[rowIndex].mountPath = mountPath;
+                          }
+
+                          setNotebookData(copiedArray);
+                        }}
+                        onDelete={() => {
+                          const copiedArray = [...notebookData];
+                          copiedArray.splice(rowIndex, 1);
+                          setNotebookData(copiedArray);
+                        }}
+                      />
+                    )}
```

Comment:
Can we extract these to functions `addNotebookData` and `deleteNotebookRow` ? we can add useCallback hook to it.